### PR TITLE
new aquarium ui and few new additions

### PR DESCRIPTION
**Pull request by Ben10Omintrix**
## About The Pull Request
gives the aquarium a new ui:
image
u can now also pet and nickname ur fish  through this interface. petting them will have them do a small dance and increase their happiness 
image

the hearts indicate how happy the fish is with the tank's living conditions and if theyve been petted recently.

## Why It's Good For The Game
gives aquariums a better UI making it easier to use

## Changelog
:cl:
qol: gives aquariums a new easier to use UI
/:cl:

### CODE REVIEW

File: code/modules/fishing/aquarium/aquarium.dm [hunk 1]
```diff
@@ -345,14 +362,17 @@
...
+			var/obj/item/fish/fish = locate(params["fish_reference"]) in contents
+			fish?.pet_fish(user)
+		if("remove_item")
+			var/atom/movable/item = locate(params["item_reference"]) in contents
+			item?.forceMove(drop_location())
+			to_chat(user, span_notice("You take out [item] from [src]."))
+		if("rename_fish")
+			var/atom/movable/fish = locate(params["fish_reference"]) in contents
+			var/chosen_name = params["chosen_name"]
+			fish.name = chosen_name
```
Comment (AnturK) [inline code/modules/fishing/aquarium/aquarium.dm:375]: Needs sanitizing

**Comment by deathrobotpunch** [comment]
I love petting fishies

**Comment by Ghommie** [comment]
Nice. I think you should make happiness affect the anger mechanic of chasm chrabs / lobstrosities if possible.

File: tgui/packages/tgui/interfaces/Aquarium.tsx [hunk 2]
```diff
@@ -17,9 +24,29 @@ type Data = {
...
+  fish_data: fish_data[];
+  prop_data: prop_data[];
   allow_breeding: BooleanLike;
   feeding_interval: number;
+  heart_icon: string;
+  heart_icon_state: string;
+  heart_empty_icon_state: string;
+};
+
+type fish_data = {
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/Aquarium.tsx:36]: types are pascal case

File: tgui/packages/tgui/interfaces/Aquarium.tsx [hunk 3]
```diff
@@ -17,9 +24,29 @@ type Data = {
...
+  fish_ref: string;
+  fish_name: string;
+  fish_happiness: number;
+  fish_icon: string;
+  fish_icon_state: string;
+  fish_health: number;
+};
+
+type prop_data = {
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/Aquarium.tsx:45]: same

File: tgui/packages/tgui/interfaces/Aquarium.tsx [hunk 4]
```diff
@@ -17,9 +24,29 @@ type Data = {
...
+  prop_ref: string;
+  prop_name: string;
+  prop_icon: string;
+  prop_icon_state: string;
 };
 
 export const Aquarium = (props) => {
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/Aquarium.tsx:52]: this component is massive, and should be broken up into more parts. components are ideally 30-100 lines each, and files are no larger than 500

File: tgui/packages/tgui/interfaces/Aquarium.tsx [hunk 5]
```diff
@@ -13,98 +18,338 @@ import { Window } from '../layouts';
...
+  fishData: fishData[];
+  propData: propData[];
+  allowBreeding: BooleanLike;
+  feedingInterval: number;
+  heartIcon: string;
+  heartIconState: string;
+  heartEmptyIconState: string;
+};
+
+type fishData = {
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/Aquarium.tsx:34]:
```suggestion
type FishData = {
```
pascal case for types

File: tgui/packages/tgui/interfaces/Aquarium.tsx [hunk 6]
```diff
@@ -13,98 +18,338 @@ import { Window } from '../layouts';
...
+  fish_ref: string;
+  fish_name: string;
+  fish_happiness: number;
+  fish_icon: string;
+  fish_icon_state: string;
+  fish_health: number;
+};
+
+type propData = {
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/Aquarium.tsx:43]: same here

**Review by jlsnow301 — CHANGES_REQUESTED** [review]

**Comment by jlsnow301** [comment]
@Ben10Omintrix in your most recent commit [335bd55],(https://github.com/tgstation/tgstation/pull/85627/commits/335bd55011b4912479b43359b74cdeb53be297d4) you also pascal cased the list name incoming from dm, but this is not reflected in the typescript side. you are instead sending two duplicate lists,  fishData & FishData (same with props) and expecting only the former.

with my ask for pascal casing the type - that's just the type, not the actual var name on either side. thus, this change was not needed and should just be `fishData` and `propData`

see the code here:
https://github.com/tgstation/tgstation/blob/master/code/modules/fishing/aquarium/aquarium.dm#L313-L328

**Comment by Ben10Omintrix** [comment]
> @Ben10Omintrix in your most recent commit [335bd55](https://github.com/tgstation/tgstation/pull/85627/commits/335bd55011b4912479b43359b74cdeb53be297d4) you also pascal cased the list name incoming from dm, but this is not reflected in the typescript side. you are instead sending two duplicate lists,  fishData & FishData (same with props) and expecting only the former.
> 
> 
> 
> with my ask for pascal casing the type - that's just the type, not the actual var name on either side. thus, this change was not needed and should just be `fishData` and `propData`
> 
> 
> 
> see the code here:
> 
> https://github.com/tgstation/tgstation/blob/master/code/modules/fishing/aquarium/aquarium.dm#L313-L328
> 
> 

whoops

**Comment by Ben10Omintrix** [comment]
> @Ben10Omintrix in your most recent commit [335bd55](https://github.com/tgstation/tgstation/pull/85627/commits/335bd55011b4912479b43359b74cdeb53be297d4) you also pascal cased the list name incoming from dm, but this is not reflected in the typescript side. you are instead sending two duplicate lists, fishData & FishData (same with props) and expecting only the former.
> 
> with my ask for pascal casing the type - that's just the type, not the actual var name on either side. thus, this change was not needed and should just be `fishData` and `propData`
> 
> see the code here: https://github.com/tgstation/tgstation/blob/master/code/modules/fishing/aquarium/aquarium.dm#L313-L328

its all fixed now

**Pull request merged**